server/ladder_service/ladder_service.py

```diff
@@ -519,6 +563,66 @@ def make_game_options(player: Player) -> GameLaunchOptions:
                     player.state = PlayerState.IDLE
                 player.write_message(msg)
 
+            if abandoning_players:
+                self._logger.info(
+                    "Players failed to connect: %s",
+                    abandoning_players
+                )
+                self.violation_service.register_violations(abandoning_players)
+
+    async def launch_match(
+        self,
+        game: LadderGame,
+        host: Player,
+        guests: list[Player],
+        make_game_options: Callable[[Player], GameLaunchOptions]
+    ):
+        # Launch the host
+        if host.lobby_connection is None:
+            raise NotConnectedError([host])
```

Comment:
Why do we make this extra check? To fail faster?

Reply:
Because otherwise we'd just get a NoneType error and it wouldn't include the information about which player caused the problem